engine/gamesys/src/gamesys/components/comp_mesh.cpp

```diff
@@ -285,8 +289,8 @@ namespace dmGameSystem
             dmHashUpdateBuffer32(&state, &texture, sizeof(texture));
         }
 
-        dmGameSystem::BufferResource* br = GetVerticesBuffer(component, component->m_Resource);
-        dmHashUpdateBuffer32(&state, &br->m_Version, sizeof(br->m_Version));
+        BufferResource* br = GetVerticesBuffer(component, resource);
+        dmHashUpdateBuffer32(&state, &br->m_NameHash, sizeof(br->m_NameHash));
```

Comment:
The hash is for batching, and as such we want to batch on the resource itself, not the content version which might be the same between the instances using different buffers.